.github/workflows/test.yml

```diff
@@ -24,13 +18,13 @@ jobs:
           - "3.9"
           - "3.10"
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Setup Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
       - name: Cache pip
-        uses: actions/cache@v2
+        uses: actions/cache@v3
```

Comment:
What if we switch to the [cache property](https://github.com/actions/setup-python#caching-packages-dependencies) bundled in `setup-python`?

Reply:
what's odd here is that we're running all the tests with tox... so all the caching in the world doesn't really help with the fact that tox goes and recreates it's own virtual environment.  It does beg the question though maybe the github actions could run without tox?

Reply:
You're right, caching doesn't really help. Concerning the running the Github actions without tox from my point of view, this change would not reduce our testing time enough to undertake those changes

Reply:
let's try it anyway 👍🏻

Reply:
@mateoflorido cool... please re-review.  Looks like this idea works

Reply:
Nice! LGTM, I'll rerun the tests to see if the pip cache kicks in

Reply:
Great, it looks like it's working and it's faster as well. 🥇